File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/CustomerDeviceOutputMapper.cs
using CustomerDeviceOutputSdk = OnlinePayments.Sdk.Domain.CustomerDeviceOutput;
using CustomerDeviceOutputDto = Business.Domain.Payments.PaymentDetails.CustomerDeviceOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class CustomerDeviceOutputMapper
{
    public static CustomerDeviceOutputDto Map(CustomerDeviceOutputSdk? response)
    {
        return new()
        {
            IpAddressCountryCode = response?.IpAddressCountryCode
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/CustomerDeviceOutputMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\CustomerDeviceOutput as CustomerDeviceOutputDto;
use OnlinePayments\Sdk\Domain\CustomerDeviceOutput as CustomerDeviceOutputSdk;

final class CustomerDeviceOutputMapper
{
    public static function mapFromSdkResponse(?CustomerDeviceOutputSdk $response): ?CustomerDeviceOutputDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new CustomerDeviceOutputDto();
        $dto->ipAddressCountryCode = $response?->ipAddressCountryCode ?? null;

        return $dto;
    }
}
